code/weapon/weapons.cpp

```diff
@@ -2328,6 +2328,36 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		if (optional_string("+Lock Range:")) {
 			stuff_float(&wip->lssm_lock_range);
 		}
+
+		if (optional_string("+Warp Effect:")) {
+			int temp;
+			if (stuff_int_optional(&temp) != 2) {
+				// We have a string to parse instead.
+				char unique_id[NAME_LENGTH];
+				memset(unique_id, 0, NAME_LENGTH);
+				stuff_string(unique_id, F_NAME, NAME_LENGTH);
+				int fireball_type = fireball_info_lookup(unique_id);
+				if (fireball_type < 0) {
+					error_display(0, "Unknown fireball [%s] to use as warp effect for LSSM weapon", unique_id);
```

Comment:
Why not throw in the name in the warning message as well? If you're Eatthepath 'an LSSM weapon' might not narrow it down much ;)